airflow/timetables/events.py

```diff
@@ -0,0 +1,97 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Iterable, Optional
+
+import numpy as np
+import pendulum
+from pendulum import DateTime
+
+from airflow.timetables.base import DagRunInfo, DataInterval, TimeRestriction, Timetable
+
+
+class EventsTimetable(Timetable):
+    """
+    Timetable that schedules DAG runs at specific listed datetimes. Suitable for
+    predictable but truly irregular scheduling such as sporting events.
+
+    :param event_dates: List of datetimes for the DAG to run at
+    :param restrict_to_events: Whether manual runs should use the most recent event or
+    the current time
+    """
+
+    description = ""
+
+    def __init__(self, event_dates: Iterable[DateTime], restrict_to_events: bool = False, presorted=False):
+
+        self.event_dates = np.array(event_dates)
+        if not presorted:
+            # For long lists this could take a while, so only want to do it once
+            self.event_dates = np.sort(self.event_dates)
+        self.restrict_to_events = restrict_to_events
+        self.description = self.summary
+
+    @property
+    def summary(self) -> str:
+        return f"{len(self.event_dates)} Events"
```

Comment:
I suspect this won't be that useful in the UI. Could you include a screenshot of the tooltip/column of the homepage with a DAG using this timetable?

Reply:
It isn't very helpful. Frankly, I don't know what to put here. I could list all the events, but I worry that could look terrible if there's a super long list (e.g. every MLB baseball game).

Reply:
Maybe the date of the next event and "X more"?

Reply:
I don't think the timetable knows which events have been processed into DAG runs. And using now() would lead to next run being inconsistent with description. Am I missing something on this?

What if I just exposed it as an optional string argument so the author could give it a meaningful name?

Reply:
Oh yes, you make a good point.

Sounds good on the name over-ride.